source/common/network/io_socket_handle_impl.cc

```diff
@@ -283,6 +283,7 @@ absl::optional<uint32_t> maybeGetPacketsDroppedFromHeader([[maybe_unused]] const
 
 Api::IoCallUint64Result IoSocketHandleImpl::recvmsg(Buffer::RawSlice* slices,
                                                     const uint64_t num_slice, uint32_t self_port,
+                                                    const UdpSaveCmsgConfig& save_cmsg_config,
```

Comment:
You may need to allocate cbuf at line290 with larger size, ideally, the config knob should specify the expected type size.

Reply:
Thanks a lot for this catch. I added a field in the config proto. cc @adisuissa @RyanTheOptimist